File: dixit-companion/src/main/java/fr/erban/dxitcompanion/db/player/PlayerConverter.java
package fr.erban.dxitcompanion.db.player;

import java.util.ArrayList;
import java.util.List;

import fr.erban.dxitcompanion.game.player.PlayerBean;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class PlayerConverter {

    public static PlayerEntity toEntity(final PlayerBean playerBean) {

        return PlayerEntity.builder()
                .name(playerBean.getName())
                .nbGames(playerBean.getNbGames())
                .nbWins(playerBean.getNbWins())
                .build();
    }

    public static PlayerBean toBean(final PlayerEntity playerEntity) {

        return PlayerBean.builder()
                .currentScore(0)
                .name(playerEntity.getName())
                .nbGames(playerEntity.getNbGames())
                .nbWins(playerEntity.getNbWins())
                .persisted(true)
                .scoresheet(new ArrayList<>())
                .build();
    }

    public static List<PlayerEntity> toEntities(final List<PlayerBean> playerBeans) {

        final List<PlayerEntity> entities = new ArrayList<>();

        for (PlayerBean playerBean : playerBeans) {
            entities.add(toEntity(playerBean));
        }

        return entities;
    }
}

File: dixit-companion/src/main/java/fr/erban/dxitcompanion/db/player/PlayerConverter.kt
package fr.erban.dxitcompanion.db.player

import fr.erban.dxitcompanion.game.player.PlayerBean

object PlayerConverter {

    fun toBean(entity: PlayerEntity): PlayerBean = PlayerBean(
        name                        = entity.name,
        currentScore                = 0,
        nbGames                     = entity.nbGames,
        nbWins                      = entity.nbWins,
        nbAsStoryteller             = entity.nbAsStoryteller,
        nbAsVoter                   = entity.nbAsVoter,
        nbFoundStoryteller          = entity.nbFoundStoryteller,
        nbStorytellerOptimalTurns   = entity.nbStorytellerOptimalTurns,
        totalPoints                 = entity.totalPoints,
        persisted                   = true,
        scoresheet                  = emptyList(),
        colorHex                    = entity.colorHex,
        emoji                       = entity.emoji
    )

    fun toEndGameEntities(players: List<PlayerBean>, winnerName: String?): List<PlayerEntity> =
        players.map { player ->
            PlayerEntity(
                name                        = player.name,
                nbGames                     = player.nbGames + 1,
                nbWins                      = player.nbWins + (if (player.name == winnerName) 1 else 0),
                nbAsStoryteller             = player.nbAsStoryteller,
                nbAsVoter                   = player.nbAsVoter,
                nbFoundStoryteller          = player.nbFoundStoryteller,
                nbStorytellerOptimalTurns   = player.nbStorytellerOptimalTurns,
                // totalPoints accumulates the score of this game on top of the historical total
                totalPoints                 = player.totalPoints + player.currentScore,
                colorHex                    = player.colorHex,
                emoji                       = player.emoji
            )
        }
}
